### PR TITLE
Script to generate gitignore file

### PR DESCRIPTION
**Pull request by rolandleth**
## Description

Calls https://gitignore.io to generate a `.gitignore` file for the given technologies and copies the result to clipboard.

## Type of change

- [x] New script command

## Screenshot

2021-03-11 at 02 36 04
Bildschirmfoto 2021-03-11 um 2 37 10 PM

- [x] I have read [Contribution Guidelines](https://github.com/raycast/script-commands/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by grzegorzkrukowski** [comment]
Thanks for the PR - that's interesting one ! 

I wonder - does it make sense to keep it as `fullOutput` mode ? It doesn't really allow user to edit anything there - also there is no information about everything being copied to clipboard - maybe having it as silent with toast could be enough for this one?

**Comment by rolandleth** [comment]
My initial idea was to display the output and copy if _after_, with a second command, so the flow would've been "generate, check if it's what you want, copy it". Funnily enough, I realised just now, while writing this, that it's possible to do just that, which is why there's a copy in the script itself 😄

I was a bit torn about this, but you only really need to check it once; from then on, you might not know _what_ you're exactly getting, but you will know that it's what you need, since you verified it in the past.

So I'll change it to silent and handle the case where it returns nothing.

File: commands/developer-utils/generate-git-ignore.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Required parameters:
+# @raycast.schemaVersion 1
+# @raycast.title Generate a .gitignore file
+# @raycast.mode fullOutput
```
Comment (grzegorzkrukowski) [inline commands/developer-utils/generate-git-ignore.sh:6]: @rolandleth I was about to merge it, but I see it's still `fullOutput` mode for this one - you mentioned in commend you want to make it silent - or you changed your mind ?

**Review by grzegorzkrukowski — CHANGES_REQUESTED** [review]

File: commands/developer-utils/generate-git-ignore.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
```
Comment (grzegorzkrukowski) [inline commands/developer-utils/generate-git-ignore.sh:2]:
```suggestion

# List of available gitignore types: https://www.toptal.com/developers/gitignore/api/list

```

File: commands/developer-utils/generate-git-ignore.sh (hunk 1 above)
Comment (grzegorzkrukowski) [inline commands/developer-utils/generate-git-ignore.sh:6]:
```suggestion
# @raycast.mode silent
```

File: commands/developer-utils/generate-git-ignore.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# Required parameters:
+# @raycast.schemaVersion 1
+# @raycast.title Generate a .gitignore file
+# @raycast.mode fullOutput
+# @raycast.packageName Developer Utilities
+
+# Optional parameters:
+# @raycast.icon 🤐
+# @raycast.argument1 { "type": "text", "placeholder": "Technologies (react,node,vscode)", "optional": true }
```
Comment (grzegorzkrukowski) [inline commands/developer-utils/generate-git-ignore.sh:11]:
```suggestion
# @raycast.argument1 { "type": "text", "placeholder": "Type (react,node,vscode)", "optional": false }
```

File: commands/developer-utils/generate-git-ignore.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+# Documentation:
+# @raycast.description Generates a .gitignore file via https://gitignore.io
+# @raycast.author Roland Leth
+# @raycast.authorURL https://runtimesharks.com
+
+params="$1"
+
+if [[ -z "$params" ]]
+then
+	params="react,node,vscode"
```
Comment (grzegorzkrukowski) [inline commands/developer-utils/generate-git-ignore.sh:22]:
```suggestion
	echo "Missing params"
	exit 1
```

File: commands/developer-utils/generate-git-ignore.sh [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+fi
+
+result=$(curl -s "https://www.toptal.com/developers/gitignore/api/$params" | sed -e "1d")
+
+if [[ "$result" =~ !!\ ERROR.+!! ]]
+then
+	echo "The requested params couldn't be found"
```
Comment (grzegorzkrukowski) [inline commands/developer-utils/generate-git-ignore.sh:29]:
```suggestion
	echo "Unsupported gitignore type"
```

**Review by grzegorzkrukowski — CHANGES_REQUESTED** [review]

File: commands/developer-utils/generate-git-ignore.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# List of available gitignore types: https://www.toptal.com/developers/gitignore/api/list
+
+# Required parameters:
+# @raycast.schemaVersion 1
+# @raycast.title Generate a .gitignore file
```
Comment (grzegorzkrukowski) [inline commands/developer-utils/generate-git-ignore.sh:7]:
```suggestion
# @raycast.title Create .gitignore
```

File: commands/developer-utils/generate-git-ignore.sh [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+# Documentation:
+# @raycast.description Generates a .gitignore file via https://gitignore.io
+# @raycast.author Roland Leth
+# @raycast.authorURL https://runtimesharks.com
+
+params="$1"
+
+if [[ -z "$params" ]]
+then
+	params="react,node,vscode"
```
Comment (grzegorzkrukowski) [inline commands/developer-utils/generate-git-ignore.sh]:
```suggestion
	echo "Missing params"
	exit 1
```

File: commands/developer-utils/generate-git-ignore.sh [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+# Required parameters:
+# @raycast.schemaVersion 1
+# @raycast.title Create .gitignore
+# @raycast.mode silent
+# @raycast.packageName Developer Utilities
+
+# Optional parameters:
+# @raycast.icon 🤐
+# @raycast.argument1 { "type": "text", "placeholder": "Type (react,node,vscode)", "optional": false }
```
Comment (grzegorzkrukowski) [inline commands/developer-utils/generate-git-ignore.sh:13]:
```suggestion
# @raycast.argument1 { "type": "text", "placeholder": "Types (react,node,vscode)", "optional": false }
```

**Pull request merged**